Essay_Analysis_Tool/Presentation/MainView.cs

```diff
@@ -447,7 +435,9 @@ private void AboutToolStripMenuItem_Click(object sender, EventArgs e)
 
         private void LoggerToolStripMenuItem_Click(object sender, EventArgs e)
         {
-            _loggerViewController.Show();
+            // This functionality will also have to be abstracted out to a service class
+            // because we do not want to mix views/controllers together.
+            //_loggerViewController.Show();
```

Comment:
With my suggestion code written before `_loggerViewController.Show();` will be correct, because the Controller already contain the service and the Controller will be subscribed to the service event so when some other Controller add new log entry the LoggerViewController will insert a new line to the View even the View not visible.

Reply:
This is where I started to get really flustered with this architecture as we are calling the controller from another view, just to perform a function that is already defined in another view. See my above comments for reference.

Reply:
Yes, sometimes to write cleaner code you will have to write a bit more but i think in the long term cleaner code will win over tightly coupled classes.